frontend/lint/tests/no-locale-with-intl-functions.unit.spec.js

```diff
@@ -0,0 +1,92 @@
+import { RuleTester } from "eslint";
+
+import noLocaleWithIntlFunctions from "../eslint-rules/no-locale-with-intl-functions";
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 2015,
+    sourceType: "module",
+    ecmaFeatures: { jsx: true },
+  },
+});
+
+const VALID_CASES = [
+  {
+    code: `'a'.localeCompare('b');`,
+  },
+  {
+    code: `new Intl.NumberFormat().format(1000);`,
+  },
+  {
+    code: `new Intl.DateTimeFormat().format(Date.now());`,
+  },
+  {
+    code: `new Intl.RelativeTimeFormat().format(3, 'seconds')`,
+  },
+  {
+    code: `new Intl.ListFormat().format(['Alice', 'Bob'])`,
+  },
+  {
+    code: `new Intl.Segmenter().segment('a')[Symbol.iterator]().next()`,
+  },
+  {
+    code: `new Intl.Collator().compare('a', 'b')`,
+  },
+];
+
+const INVALID_CASES = [
```

Comment:
It might be good to add some cases that are more similar to the P1 described. ie. where the locale is passed as a reference, not a string literal.

```js
const locale = "pt_BR"
"a".localCompare("b", locale)
```